cmd/kube-proxy/app/server.go

```diff
@@ -126,15 +131,25 @@ func NewProxyConfig() *ProxyServerConfig {
 
 func NewProxyServer(
 	config *ProxyServerConfig,
+	client *kubeclient.Client,
+	endpointsConfig *proxyconfig.EndpointsConfig,
+	endpointsHandler proxyconfig.EndpointsConfigHandler,
 	iptInterface utiliptables.Interface,
+	oomAdjuster *oom.OOMAdjuster,
 	proxier proxy.ProxyProvider,
 	recorder record.EventRecorder,
+	serviceConfig *proxyconfig.ServiceConfig,
 ) (*ProxyServer, error) {
 	return &ProxyServer{
-		Config:       config,
-		IptInterface: iptInterface,
-		Proxier:      proxier,
-		Recorder:     recorder,
+		Client:           client,
+		Config:           config,
+		EndpointsConfig:  endpointsConfig,
```

Comment:
I'm in favor of dependency injection, I just don't see these injected values being used anywhere.